File: Lab4/src/com/company/domain/Matrix.java
package com.company.domain;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Matrix {

    private List<List<Integer>> values;
    private int rows, columns;

    public Matrix(int rows, int columns) {
        this.rows = rows;
        this.columns = columns;

        Random random = new Random();

        this.values = new ArrayList<List<Integer>>(this.rows);
        for (int i = 0; i < this.rows; i ++) {
            this.values.add(new ArrayList<Integer>(this.columns));

            for (int j = 0; j < this.columns; j ++) {
                this.values.get(i).add(random.nextInt(5) + 1);
            }
        }
    }

    public int getColumnsNumber() {
        return this.columns;
    }

    public int getRowsNumber() {
        return this.rows;
    }

    public void setCellValue(int row, int col, int value){
        this.values.get(row).set(col, value);
    }

    public int getCellValue(int row, int col){
        return this.values.get(row).get(col);
    }

    public void setAllCellsToZero() {
        for (int i = 0; i < this.rows; i++) {
            for (int j = 0; j < this.columns; j++) {
                this.setCellValue(i, j, 0);
            }
        }
    }

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder();
        for (int i = 0; i < this.rows; i++){
            stringBuilder.append(this.values.get(i).toString()).append("\n");
        }

        return stringBuilder.toString();
    }
}

File: Lab3/Lab3/Matrix.cpp
//
//  Matrix.cpp
//  Lab3
//
//  Created by Teofana Moisi on 27/10/2019.
//  Copyright © 2019 Teofana Moisi. All rights reserved.
//

#include "Matrix.hpp"

Matrix::Matrix() {
    this->rows = 0;
    this->columns = 0;
}

Matrix::Matrix(int rows, int columns) : values(rows, vector<int>(columns)) {
    this->rows = rows;
    this->columns = columns;
    
    for (int i = 0; i < this->rows; i++) {
        for (int j = 0; j < this->columns; j++) {
            int random = rand() % 400 + 100;
            this->values[i][j] = random;
        }
    }
}

Matrix::~Matrix() {
}

string Matrix::printMatrix() {
    string result = "";
    
    for (vector<int> vec: values) {
        for (int val: vec) {
            result += to_string(val) + ' ';
        }
        result += '\n';
    }
    
    return result;
}

int Matrix::getRowsNumber() {
    return this->rows;
}

int Matrix::getColumnsNumber() {
    return this->columns;
}

void Matrix::setCellValue(int row, int column, int value) {
    this->values[row][column] = value;
}

int Matrix::getCellValue(int row, int column) {
    return this->values[row][column];
}
